### PR TITLE
Moved mass grave terrain and special to map extra

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: Infrastructure "Moved mass grave terrain and special to map extra"```

#### Purpose of change

Get rid of some 1x1x1 overmap specials to  free up special slots for bigger and more interesting locations.

#### Describe the solution

Moved mass grave overmap terrain and special to map extra using `update_mapgen` function.

#### Additional context

I am not sure if I should apply a better migration for obsolete terrains (applying extras, count neighboring terrains when choosing migration terrain, etc).

There are more 1x1x1 specials that could be moved out of special (ponds, basins, etc).

We would eventually need to allow:

- random rotation of `update_mapgen` map extras;
- multitile map extras to make possible moving bigger overmap specials to map extras (wasp nests, etc).

### CODE REVIEW

**Comment by Inglonias** [comment]
On one hand this is probably the correct thing to do. On the other hand, this makes it easier to accidentally trip over mass graves, which I don't think is something people want to do.

**Comment by ZhilkinSerg** [comment]
> 
> 
> On one hand this is probably the correct thing to do. On the other hand, this makes it easier to accidentally trip over mass graves, which I don't think is something people want to do.

I don't see how - map extras are tracked on the map.

**Comment by Inglonias** [comment]
> 
> 
> > On one hand this is probably the correct thing to do. On the other hand, this makes it easier to accidentally trip over mass graves, which I don't think is something people want to do.
> 
> I don't see how - map extras are tracked on the map.

They are indeed, but that's turned off by default, I believe. As I said earlier, I don't think this complaint is enough to justify not doing this. I just wanted to make sure you were aware of that.

File: data/json/overmap/map_extras.json [hunk 1]
```diff
@@ -387,5 +387,15 @@
...
     "description": "A prison bus.",
     "generator": { "generator_method": "update_mapgen", "generator_id": "mx_prison_bus" }
+  },
+  {
+    "id": "mx_mass_grave",
+    "type": "map_extra",
+    "name": "Mass Grave",
+    "description": "A mass grave.",
+    "generator": { "generator_method": "update_mapgen", "generator_id": "mx_mass_grave" },
+    "sym": "x",
```
Comment (anothersimulacrum) [inline data/json/overmap/map_extras.json]: To match the previous overmap tile.
```suggestion
    "sym": "X",
```

**Comment by Amoebka** [comment]
Is it possible/feasible to increase the overmap special count per overmap at some point? On one hand, small specials taking up slots is an issue, but on the other, map extras are basically strictly inferiour specials. Autonoting is not a valid alternative to having "specials"/extras properly displayed on the map, search functions are duplicated, and it's just an overall duplication of features.

A much better long-term solution would be to actually convert extras (where possible, like ponds, sheds, etc) to terrains and increase the overmap specials limit accordingly.

**Comment by I-am-Erk** [comment]
Extras aren't "inferior" specials. They have a totally different spawning mechanism and are used for different things.

File: data/json/obsolete_terrains.json [hunk 2]
```diff
@@ -297,6 +297,10 @@
       "house_base_south",
       "house_base_east",
       "house_base_west",
+      "mass_grave_north",
+      "mass_grave_south",
+      "mass_grave_east",
+      "mass_grave_west"
```
Comment (ZhilkinSerg) [inline data/json/obsolete_terrains.json:303]:
```suggestion
      "mass_grave_west",
```

**Pull request merged**